src/ngx_http_lua_uri.c

```diff
@@ -55,12 +56,19 @@ ngx_http_lua_ngx_req_set_uri(lua_State *L)
         return luaL_error(L, "attempt to use zero-length uri");
     }
 
-    if (ngx_http_lua_check_unsafe_string(r, p, len, "uri") != NGX_OK) {
-        return luaL_error(L, "attempt to set unsafe uri");
+
+    if (n >= 3) {
+        luaL_checktype(L, 3, LUA_TBOOLEAN);
+        binary = lua_toboolean(L, 3);
     }
 
-    if (n == 2) {
+    if (!binary 
```

Comment:
these lines should move to line 62, inside the previous {} block.

Reply:
without binary argument， we also need to check unsafe_string